cpp/src/io/orc/stripe_data.cu

```diff
@@ -642,6 +642,18 @@ static const __device__ __constant__ uint8_t kRLEv2_W[32] = {
   1,  2,  3,  4,  5,  6,  7,  8,  9,  10, 11, 12, 13, 14, 15, 16,
   17, 18, 19, 20, 21, 22, 23, 24, 26, 28, 30, 32, 40, 48, 56, 64};
 
+/**
+ * @brief Maps the RLEv2 patch size (pw + pgw) to number of bits
+ *
+ * Patch size (in bits) is only allowed to be from the below set. If `pw + pgw == 34` then the size
+ * of the patch in the file is the smallest size in the set that can fit 34 bits i.e.
+ * `ClosestFixedBitsMap[34] == 40`
+ */
+static const __device__ __constant__ uint8_t ClosestFixedBitsMap[65] = {
```

Comment:
Is it possible to initialize this with a function so that the rules are evident in the code?

Reply:
This is taken from https://github.com/apache/orc/blob/master/c%2B%2B/src/RLEV2Util.cc#L30-L37 so I don't actually know the rule. The set of fixed bit widths and how they're calculated is not in the docs.

Reply:
Then it's fine as an array. Would be great if you could include a link to the source in some way.